File: lib/src/main/java/uk/iatom/rfm69/registers/PayloadLength.java
package uk.iatom.rfm69.registers; import uk.iatom.rfm69.IRegisterValue;

public enum PayloadLength implements IRegisterValue {
    
    // RegPayloadLength),
    PAYLOADLENGTH_VALUE(0x40); // Default),;

    private byte val;
    
    PayloadLength(int val) {
        this.val = (byte) val;
    }

    @Override public byte val() {
        return val;
    }
}

File: lib/src/main/java/uk/iatom/rfm69/register/values/PayloadLength.kt
package uk.iatom.rfm69.register.values

import uk.iatom.rfm69.register.IRegisterValue

enum class PayloadLength(`val`: Int): IRegisterValue { // RegPayloadLength),
    PAYLOADLENGTH_VALUE(0x40);


    // Default),;
    private val `val`: Byte

    init {
        this.`val` = `val`.toByte()
    }

    override fun `val`(): Byte {
        return `val`
    }
}
